### PR TITLE
job file: add set -o pipefail for bash jobs

### LINKED ISSUE #1783: job scripts trap pipe problem
Consider a task with a `script`:

```sh
echo "$GARBAGE" | tee 'whatever.txt'
```

The task will attempt to report both failure and succeeded. The 1st part of the pipe will fail with an unbound variable, which triggers the exit trap to send a failure message back to the suite. The 2nd part of the pipe will succeed and the script will some how continue to run before sending a succeeded message back to the suite.

This is a well known problem in bash, but we should probably document the problem in the cylc user guide?

### PR DESCRIPTION
**Pull request by matthewrmshin**
Close #1783.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
(Just need to figure out the best place to document this.)

**Comment by matthewrmshin** [comment]
@hjoliver @arjclark please review.

File: doc/cug.tex [hunk 1]
```diff
@@ -7468,6 +7468,31 @@ \subsection{No Implicit Creation of Tasks by Offset Triggers}
...
+\section{Known Issues}
+\label{KnownIssues}
+
+\subsection{Current Known Issues}
+\label{CurrentKnownIssues}
+
+The best place to find current known issues is on Github:
+\url{https://github.com/cylc/cylc/issues}.
+
+\subsection{Other Known Issues}
```
Comment (arjclark) [inline doc/cug.tex]: Could we name this "Notable Known Issues" - that way we can use the section for documenting things like the Family triggering memory overhead and the GraphViz GTK memory leak.

Comment (matthewrmshin) [inline doc/cug.tex]: Noted and modified.

File: doc/cug.tex [hunk 2]
```diff
@@ -7468,6 +7468,35 @@ \subsection{No Implicit Creation of Tasks by Offset Triggers}
...
+\label{NotableKnownIssues}
+
+\subsubsection{Use of pipes in job scripts}
+\label{PipeInJobScripts}
+
+In bash, the return status of a pipeline is normally the exit status of the
+last command. This is unsafe, because if any command in the pipeline fails, the
+script will continue nevertheless.
+
+For safetty, a cylc task job script running in bash will have the
```
Comment (hjoliver) [inline doc/cug.tex]: "safety"

**Comment by hjoliver** [comment]
Good, one doc typo.

File: doc/cug.tex (hunk 2 above)
Comment (matthewrmshin) [inline doc/cug.tex]: Fixed.

**Pull request merged**